{
    "url": "https://api.github.com/repos/spack/spack/issues/5156",
    "repository_url": "https://api.github.com/repos/spack/spack",
    "labels_url": "https://api.github.com/repos/spack/spack/issues/5156/labels{/name}",
    "comments_url": "https://api.github.com/repos/spack/spack/issues/5156/comments",
    "events_url": "https://api.github.com/repos/spack/spack/issues/5156/events",
    "html_url": "https://github.com/spack/spack/issues/5156",
    "id": 251235566,
    "node_id": "MDU6SXNzdWUyNTEyMzU1NjY=",
    "number": 5156,
    "title": "spack info: Spurious failures on Travis",
    "user": {
        "login": "alalazo",
        "id": 4199709,
        "node_id": "MDQ6VXNlcjQxOTk3MDk=",
        "avatar_url": "https://avatars.githubusercontent.com/u/4199709?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/alalazo",
        "html_url": "https://github.com/alalazo",
        "followers_url": "https://api.github.com/users/alalazo/followers",
        "following_url": "https://api.github.com/users/alalazo/following{/other_user}",
        "gists_url": "https://api.github.com/users/alalazo/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/alalazo/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/alalazo/subscriptions",
        "organizations_url": "https://api.github.com/users/alalazo/orgs",
        "repos_url": "https://api.github.com/users/alalazo/repos",
        "events_url": "https://api.github.com/users/alalazo/events{/privacy}",
        "received_events_url": "https://api.github.com/users/alalazo/received_events",
        "type": "User",
        "site_admin": false
    },
    "labels": [
        {
            "id": 456341797,
            "node_id": "MDU6TGFiZWw0NTYzNDE3OTc=",
            "url": "https://api.github.com/repos/spack/spack/labels/tests",
            "name": "tests",
            "color": "b60205",
            "default": false,
            "description": "General test capability(ies)"
        },
        {
            "id": 671550628,
            "node_id": "MDU6TGFiZWw2NzE1NTA2Mjg=",
            "url": "https://api.github.com/repos/spack/spack/labels/travis",
            "name": "travis",
            "color": "bfdadc",
            "default": false,
            "description": null
        }
    ],
    "state": "closed",
    "locked": false,
    "assignee": null,
    "assignees": [],
    "milestone": null,
    "comments": 4,
    "created_at": "2017-08-18T12:50:22Z",
    "updated_at": "2017-08-20T22:34:35Z",
    "closed_at": "2017-08-20T22:34:35Z",
    "author_association": "MEMBER",
    "active_lock_reason": null,
    "body": "Travis is failing spuriously when trying to execute:\r\n```python\r\nspack info openmpi\r\n```\r\nduring unit tests. The stacktrace we get back is something like:\r\n```\r\n__________________________ test_it_just_runs[openmpi] __________________________\r\npkg = 'openmpi'\r\n    @pytest.mark.parametrize('pkg', [\r\n        'openmpi',\r\n        'trilinos',\r\n        'boost',\r\n        'python',\r\n        'dealii'\r\n    ])\r\n    def test_it_just_runs(pkg):\r\n>       info(pkg)\r\n../../../lib/spack/spack/test/cmd/info.py:40: \r\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \r\n../../../lib/spack/spack/main.py:406: in __call__\r\n    self.command, self.parser, args, unknown)\r\n../../../lib/spack/spack/main.py:355: in _invoke_spack_command\r\n    return_val = command(parser, args)\r\n../../../lib/spack/spack/cmd/info.py:241: in info\r\n    print_text_info(pkg)\r\n../../../lib/spack/spack/cmd/info.py:231: in print_text_info\r\n    when.colorized(), ', '.join(s.colorized() for s in specs)\r\n../../../lib/spack/spack/spec.py:2670: in colorized\r\n    return colorize_spec(self)\r\n../../../lib/spack/spack/spec.py:240: in colorize_spec\r\n    return colorize(re.sub(_separators, insert_color(), str(spec)) + '@.')\r\n/home/travis/virtualenv/python3.5.3/lib/python3.5/re.py:182: in sub\r\n    return _compile(pattern, flags).sub(repl, string, count)\r\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \r\nself = <spack.spec.colorize_spec.<locals>.insert_color object at 0x7fd970749550>\r\nmatch = <_sre.SRE_Match object; span=(0, 1), match='o'>\r\n    def __call__(self, match):\r\n        # ignore compiler versions (color same as compiler)\r\n        sep = match.group(0)\r\n        if self.last == '%' and sep == '@':\r\n            return cescape(sep)\r\n        self.last = sep\r\n    \r\n>       return '%s%s' % (color_formats[sep], cescape(sep))\r\nE       KeyError: 'o'\r\n../../../lib/spack/spack/spec.py:238: KeyError\r\n```\r\nThe problem first occurred in #4994, and after trying without success to reproduce the issue both on Travis and locally I discarded it as some strange environment problem (like the ones we get from time to time regarding `svn` failing). Unfortunately, it happened again when building on `develop` [in build 12355](https://travis-ci.org/LLNL/spack/builds/265784067). Restarting the two failing builds made the issue disappear.",
    "performed_via_github_app": null
}